modules/test/jenkins-results-parser/src/main/java/com/liferay/jenkins/results/parser/spira/SpiraRelease.java

```diff
@@ -33,6 +36,119 @@
  */
 public class SpiraRelease {
 
+	public static SpiraRelease createSpiraRelease(
+			SpiraProject spiraProject, String releaseName)
+		throws IOException {
+
+		return createSpiraRelease(spiraProject, releaseName, null);
+	}
+
+	public static SpiraRelease createSpiraRelease(
+			SpiraProject spiraProject, String releaseName,
+			Integer parentReleaseId)
+		throws IOException {
+
+		String urlPath = "projects/{project_id}/releases/{parent_release_id}";
+
+		Map<String, String> urlPathReplacements = new HashMap<>();
+
+		urlPathReplacements.put(
+			"parent_release_id", String.valueOf(parentReleaseId));
+		urlPathReplacements.put(
+			"project_id", String.valueOf(spiraProject.getID()));
+
+		if ((parentReleaseId == null) || (parentReleaseId == 0)) {
+			urlPath = urlPath.replace("/{parent_release_id}", "");
+		}
+
+		JSONObject requestJSONObject = new JSONObject();
+
+		requestJSONObject.put(
+			"Name", StringEscapeUtils.unescapeJava(releaseName));
+		requestJSONObject.put("ReleaseStatusId", SpiraRelease.STATUS_PLANNED);
+		requestJSONObject.put("ReleaseTypeId", SpiraRelease.TYPE_MAJOR_RELEASE);
+
+		Calendar calendar = Calendar.getInstance();
+
+		requestJSONObject.put("StartDate", _toDateString(calendar));
+
+		calendar.add(Calendar.MONTH, 1);
+
+		requestJSONObject.put("EndDate", _toDateString(calendar));
+
+		JSONObject responseJSONObject = SpiraRestAPIUtil.requestJSONObject(
+			urlPath, null, urlPathReplacements, HttpRequestMethod.POST,
+			requestJSONObject.toString());
+
+		return spiraProject.getSpiraReleaseByID(
+			responseJSONObject.getInt("ReleaseId"));
+	}
+
+	public static SpiraRelease createSpiraReleaseByPath(
+			SpiraProject spiraProject, String releasePath)
+		throws IOException {
+
+		List<SpiraRelease> spiraReleases = spiraProject.getSpiraReleasesByPath(
+			releasePath);
+
+		if (!spiraReleases.isEmpty()) {
+			return spiraReleases.get(0);
+		}
+
+		String[] releasePathNames = releasePath.split("(?<!\\\\)\\/");
+
+		StringBuilder sb = new StringBuilder();
+
+		for (int i = 1; i < (releasePathNames.length - 1); i++) {
+			sb.append("/");
+			sb.append(releasePathNames[i]);
+		}
+
+		String releaseName = releasePathNames[releasePathNames.length - 1];
+
+		String parentReleasePath = sb.toString();
+
+		if (parentReleasePath.isEmpty()) {
+			return createSpiraRelease(spiraProject, releaseName);
+		}
+
+		SpiraRelease parentSpiraRelease = createSpiraReleaseByPath(
+			spiraProject, parentReleasePath);
```

Comment:
Shouldn't the `parentReleasePath` drop the last item in `releasePathNames[]`, not the first?
```
   		for (int i = 0; i < (releasePathNames.length - 2); i++) {
			sb.append("/");
			sb.append(releasePathNames[i]);
		}
```